### PR TITLE
Issue #2393 - hzn dev service log should have an argument for the URL of the service to display

### PR DESCRIPTION
**Pull request by Isabel-Gan**
Signed-off-by: Isabel Gan <Isabel.Gan@ibm.com>

### CODE REVIEW

File: cli/hzn.go [hunk 1]
```diff
@@ -232,7 +235,7 @@ Environment Variables:
 
 	devServiceCmd := devCmd.Command("service", msgPrinter.Sprintf("For working with a service project."))
 	devServiceLogCmd := devServiceCmd.Command("log", msgPrinter.Sprintf("Show the container/system logs for a service."))
-	devServiceLogCmdServiceName := devServiceLogCmd.Flag("service", msgPrinter.Sprintf("The name of the service whose log records should be displayed. The service name is the same as the url field of a service definition.")).Short('s').String()
```
Comment (linggao) [inline cli/hzn.go:235]: Should keep this flag unchange, do not remove it. Issue https://github.com/open-horizon/anax/pull/2387 will change the flag to `-c`.

File: cli/dev/service.go [hunk 2]
```diff
@@ -239,6 +242,36 @@ func ServiceValidate(homeDirectory string, userInputFile string, configFiles []s
...
+			return res, nil
+		}
+	}
+
+	// unsuccessful search
+	return nil, nil
+
+}
+
 func ServiceLog(homeDirectory string, serviceName string, tailing bool) {
```
Comment (linggao) [inline cli/dev/service.go:275]: This should be changed to 
````
func ServiceLog(homeDirectory string, serviceName string, containerName string, tailing bool)
````
Replacing serviceName with containerName in old code.

File: cli/dev/service.go [hunk 3]
```diff
@@ -252,28 +285,21 @@ func ServiceLog(homeDirectory string, serviceName string, tailing bool) {
 		cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, "'%v %v' %v", SERVICE_COMMAND, SERVICE_LOG_COMMAND, wderr)
 	}
 
-	// Get the deployment config. This is a top-level service because it's the one being launched, so it is treated as
-	// if it is managed by an agreement.
-	dc, _, cerr := serviceDef.ConvertToDeploymentDescription(true)
-	if cerr != nil {
-		cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, "'%v %v' %v", SERVICE_COMMAND, SERVICE_LOG_COMMAND, cerr)
```
Comment (linggao) [inline cli/dev/service.go:259]: Still need this code to locate the container once the service def is found. You can move this to after the searchDependencies

**Review by linggao — CHANGES_REQUESTED** [review]
`hzn dev service log dep1` did not work for me, where dep1 is the service url for dependent service.

File: cli/dev/service.go (hunk 3 above)
Comment (Isabel-Gan) [reply cli/dev/service.go:259]: Addressed this in recent changes

File: cli/dev/service.go [hunk 4]
```diff
@@ -252,28 +285,21 @@ func ServiceLog(homeDirectory string, serviceName string, tailing bool) {
...
-			cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("'%v %v' More than one service has been found for deployment. Please specify the service name by -s flag", SERVICE_COMMAND, SERVICE_LOG_COMMAND))
-		}
-
-		// Apply service name
-		for name, svc := range dc.Services {
-			serviceName = name
-			if svc.LogDriver != "" {
-				logDriver = svc.LogDriver
-			}
-		}
```
[the remark this answers is not in the document]
Comment (Isabel-Gan) [reply cli/dev/service.go:274]: Addressed this in recent changes

File: cli/hzn.go (hunk 1 above)
Comment (Isabel-Gan) [reply cli/hzn.go:235]: Addressed this in recent changes

**Pull request closed without merging**